client/components/advanced-filters/index.js

```diff
@@ -0,0 +1,215 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { Component, Fragment } from '@wordpress/element';
+import { SelectControl, Button, FormTokenField, Dropdown } from '@wordpress/components';
+import { partial, findIndex, find, difference } from 'lodash';
+import PropTypes from 'prop-types';
+import Gridicon from 'gridicons';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import './style.scss';
+
+class AdvancedFilters extends Component {
+	constructor( props ) {
+		super( props );
+		this.state = {
+			match: props.config.matches[ 0 ],
+			activeFilters: [
+				/**
+				 * Example activeFilter
+				 * { key: ‘product’, rule: ‘includes’, value: [ ‘one’, ‘two’ ] }
+				 */
+			],
+		};
+
+		this.onMatchChange = this.onMatchChange.bind( this );
+		this.onFilterChange = this.onFilterChange.bind( this );
+		this.getSelector = this.getSelector.bind( this );
+		this.getAvailableFilterKeys = this.getAvailableFilterKeys.bind( this );
+		this.addFilter = this.addFilter.bind( this );
+		this.removeFilter = this.removeFilter.bind( this );
+		this.clearAll = this.clearAll.bind( this );
```

Comment:
should this be `clearAllFilters` to match the `addFilter` and `removeFilter` above?